src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+            statement.setString(TITLE_COLUMN, book.getTitle());
+            statement.setBigDecimal(PRICE_COLUMN, book.getPrice());
+            if (statement.executeUpdate() > MIN_EXECUTE_UPDATE) {
+                ResultSet resultSet = statement.executeQuery();
+                if (resultSet.next()) {
+                    book.setId(resultSet.getLong(ID));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create book", e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(ID_COLUMN, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString(TITLE);
+                BigDecimal price = BigDecimal.valueOf(resultSet.getInt(PRICE));
+                Book book = new Book(id, title, price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot book with id" + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                String title = resultSet.getString(TITLE_COLUMN);
+                BigDecimal price = resultSet.getBigDecimal(PRICE_COLUMN);
+                Long id = resultSet.getObject(ID, Long.class);
+                Book book = new Book(id, title, price);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(ID_COLUMN, book.getId());
+            statement.setString(TITLE_COLUMN, book.getTitle());
+            statement.setBigDecimal(PRICE_COLUMN, book.getPrice());
+            if (statement.executeUpdate() > MIN_EXECUTE_UPDATE) {
+                return book;
+            } else {
+                throw new DataProcessingException("Cannot update book with id"
```

Comment:
Use appropriate message
```suggestion
                throw new DataProcessingException("Expected to update at least 1 row, but 0 was updated."
```